javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionClassLoader.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+
+/**
+ * This classloader is used to load arbitrary extensions for Otel Java instrumentation agent. Such
+ * extensions may include SDK components (exporters or propagators) and additional instrumentations.
+ * They have to be isolated and shaded to reduce interference with the user application and to make
+ * it compatible with shaded SDK used by the agent.
+ */
+// TODO find a way to initialize logging before using this class
+public class ExtensionClassLoader extends URLClassLoader {
+
+  public static ClassLoader getInstance(ClassLoader parent) {
+    // TODO add support for old deprecated properties, otel.exporter.jar and otel.initializer.jar
+    // TODO add support for system properties
+    URL extension = parseLocation(System.getenv("OTEL_JAVAAGENT_EXTENSIONS"));
+    if (extension != null) {
+      try {
+        URL wrappedUrl = new URL("otel", null, -1, "/", new RemappingUrlStreamHandler(extension));
```

Comment:
instead of using a custom `URLStreamHandler` did you consider alternatives like doing the transform inside a `ClassFileTransformer` or `findClass`?

Reply:
The alternative approach https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2774 renames in the `findClass()` method

Reply:
We want to remap resources as well, so we cannot use class file transformers, I think...